src/test/scala/firrtlTests/CheckCombLoopsSpec.scala

```diff
@@ -0,0 +1,153 @@
+// See LICENSE for license details.
+
+package firrtlTests
+
+import firrtl._
+import firrtl.ir._
+import firrtl.passes._
+import firrtl.Mappers._
+import annotations._
+
+class CheckCombLoopsSpec extends SimpleTransformSpec {
+
+  class TestCombLoopsCheck extends PassBasedTransform {
+    def inputForm = LowForm
+    def outputForm = LowForm
+    def passSeq = Seq(passes.CheckCombLoops)
+  }
+
+  def emitter = new LowFirrtlEmitter
+
+  def transforms = Seq(
+    new ChirrtlToHighFirrtl,
+    new IRToWorkingIR,
+    new ResolveAndCheck,
+    new HighFirrtlToMiddleFirrtl,
+    new MiddleFirrtlToLowFirrtl,
+    new TestCombLoopsCheck
+  )
+
+  "Simple combinational loop" should "throw an exception" in {
+    val input = """
+circuit hasloops :
```

Comment:
stripMargin lets you add `|` before the text, look at other tests, e.g. UnitTests.scala. Fix this for all your tests.